python/cudnn/gemm/cutedsl/grouped/wgrad/_bf16_api.py

```
@@ -175,23 +184,27 @@ def _validate_offsets_once(self, offsets: torch.Tensor, *, tokens_sum: int) -> N
+        if get_data_ptr(tensor) % 8 != 0:
             raise ValueError("wgrad_ptrs data pointer must be 8-byte aligned")
 
     @staticmethod
     def _validate_data_alignment(tensor: torch.Tensor, name: str, alignment: int = 16) -> None:
-        if tensor.data_ptr() % alignment != 0:
+        if get_data_ptr(tensor) % alignment != 0:
             raise ValueError(f"{name} data pointer must be {alignment}-byte aligned")
 
-    @staticmethod
-    def _record_pointer_stream(pointers: torch.Tensor, current_stream: cuda.CUstream) -> None:
+    def _record_pointer_stream(self, pointers: torch.Tensor, current_stream: cuda.CUstream) -> None:
```

Ruff (0.16.1)

[error] Undefined name `torch`

(F821)